app/routes.js

```diff
@@ -14,6 +16,16 @@ exports.init = app => {
     paramsValidator.validateSchemaAndFail(schemas.users.create),
     usersController.create
   );
+  app.post(
+    `${URL}/admin/users`,
+    [
+      paramsValidator.validateSchemaAndFail(schemas.admins.create),
+      adminValidator.create,
+      adminValidator.validateAdmin,
+      authorizationValidator.validate
+    ],
```

Comment:
I think you need only two middlewares, one for check the schema and another for the token and the permission necessary